cleverhans/experimental/certification/nn.py

```diff
@@ -0,0 +1,157 @@
+    self.layer_types = []
+    self.sizes = []
+    # Setting the sizes of the layers of the network
+    # sizes[i] contains the size of x_i
+    for i in range(self.num_hidden_layers):
+      shape = np.shape(net_weights[i])
+      self.sizes.append(int(shape[1]))
+      self.weights.append(
+          tf.convert_to_tensor(net_weights[i], dtype=tf.float32))
+      self.biases.append(tf.convert_to_tensor(net_biases[i], dtype=tf.float32))
+      self.layer_types.append(net_layer_types[i])
+
+    # Last layer shape
+    self.sizes.append(int(np.shape(net_weights[self.num_hidden_layers - 1])[0]))
+    self.final_weights = tf.convert_to_tensor(
+        net_weights[self.num_hidden_layers], dtype=tf.float32)
+    self.final_bias = tf.convert_to_tensor(
+        net_biases[self.num_hidden_layers], dtype=tf.float32)
+
+  def forward_pass(self, vector, layer_index, is_transpose=False, is_abs=False):
+    """Performs forward pass through the layer weights at layer_index.
+
+    Args:
+      vector: vector that has to be passed through in forward pass
+      layer_index: index of the layer
+      is_transpose: whether the weights of the layer have to be transposed
+      is_abs: whether to take the absolute value of the weights
+
+    Returns:
+      tensor that corresponds to the forward pass through the layer
+    Raises:
+      ValueError: if the layer_index is negative or more than num hidden layers
+    """
+    if (layer_index < 0 or layer_index > self.num_hidden_layers):
+      raise ValueError('Invalid layer index')
+
+    if (self.layer_types[layer_index] == 'ff' or
+        self.layer_types[layer_index] == 'ff_relu'):
+      if is_transpose:
+        return tf.matmul(tf.transpose(self.weights[layer_index]), vector)
+      elif is_abs:
+        return tf.matmul(tf.abs(self.weights[layer_index]), vector)
+      else:
+        return tf.matmul(self.weights[layer_index], vector)
+
+    raise NotImplementedError('Unsupported layer type: {0}'.format(
+        self.layer_types[layer_index]))
+
+def load_network_from_checkpoint(checkpoint, model_json):
+  """Function to read the weights from checkpoint based on json description.
+
+    Args:
+      checkpoint: tensorflow checkpoint with trained model to
+        verify
+      model_json: path of json file with model description of
+        the network list of dictionary items for each layer
+        containing 'type', 'weight_var', 'bias_var' and
+        'is_transpose' 'type'is one of {'ff', 'ff_relu' or
+        'conv'}; 'weight_var' is the name of tf variable for
+        weights of layer i; 'bias_var' is the name of tf
+        variable for bias of layer i; 'is_transpose' is set to
+        True if the weights have to be transposed as per
+        convention Note that last layer is always feedforward
+        (verification operates at the layer below the final
+        softmax for more numerical stability)
+
+    Returns:
+      net_weights: list of numpy matrices of weights of each
+      layer
```

Comment:
There are still some issues with formatting of comment:
* `layer` should have proper indentation (need to add two spaces). Also I think `layer` probably can be moved to the previous line and you still should be within line length limit
* unnecessary `[` in front of `convertion` (next line), and also this line should have proper indentation.